src/builder.rs

```diff
@@ -140,6 +142,8 @@ pub struct MetaBuilder {
     service_handle: ProtocolHandle<Box<dyn ServiceProtocol + Send + 'static>>,
     session_handle: SessionHandleFn,
     select_version: SelectVersionFn,
+    before_send: Option<Box<dyn Fn(bytes::Bytes) -> bytes::Bytes + Send + 'static>>,
```

Comment:
What's the choice here that `before_send` is an `Option`, but `before_receive` is an `Box<Fn>`?

Reply:
Because the message has a unified entry, only one fn can be used. 

The received is that each protocol in each session has an entry and there is no aggregation, so here need to have a handler that can always generate the exact same function, `fn -> Option<fn>`.

Each time a protocol open, get a `receive fn` to it's stream